collector/node_collector.go

```diff
@@ -110,43 +112,39 @@ func (h *DCOSHost) RunPoller() {
 func (h *DCOSHost) pollHost() metricsMeta {
 	metrics := metricsMeta{}
 	now := time.Now().UTC()
-	nodeColLog.Infof("Polling the DC/OS host at %s:%d. Started at %s", h.IPAddress, h.Port, now.String())
+	metrics.timestamp = now.Unix()
+
+	// Fetch node-level metrics for all DC/OS roles
+	nodeColLog.Debugf("Fetching node-level metrics from DC/OS host %s:%d", h.IPAddress, h.Port)
+	nm, err := h.getNodeMetrics()
+	if err != nil {
+		nodeColLog.Errorf("Failed to get node-level metrics. %s", err)
+	} else {
+		metrics.nodeMetrics = nm
+	}
```

Comment:
Yes, we're gathering metrics from HTTP endpoints and host level stuff here. If one fails, we shouldn't block the others and we shouldn't bail out or panic. It's best effort.